### PR TITLE
improve get balance ux

### PR DESCRIPTION
**Pull request by felipemadero**
## Why this should be merged
- add flag for node id (used in combination with l1 name)
- prompt the user to choose among known l1s
- prompt the user to choose among validators

### CODE REVIEW

File: cmd/validatorcmd/getBalance.go [hunk 1]
```diff
@@ -58,68 +71,137 @@ func getBalance(_ *cobra.Command, _ []string) error {
...
-		ux.Logger.PrintToUser("  Validator Balance: %.5f", float64(balance)/float64(units.Avax))
+	validationID, cancel, err := getNodeValidationID(network, l1, nodeIDStr, validationIDStr)
+	if err != nil {
+		return err
+	}
+	if cancel {
 		return nil
 	}
+	if validationID == ids.Empty {
+		return fmt.Errorf("the specified node is not a L1 validator")
```
Comment (arturrez) [inline cmd/validatorcmd/getBalance.go:82]: nit: add `as validationID is empty` to the error msg
pls feel free to ignore this

File: cmd/validatorcmd/getBalance.go [hunk 2]
```diff
@@ -58,68 +71,137 @@ func getBalance(_ *cobra.Command, _ []string) error {
...
+
+	return nil
+}
+
+func getNodeValidationID(
+	network models.Network,
+	l1 string,
+	nodeIDStr string,
+	validationIDStr string,
+) (ids.ID, bool, error) {
```
Comment (arturrez) [inline cmd/validatorcmd/getBalance.go:99]: nit: 
it's not not clear what `bool` in return means.
pls change func name or add comment for this

File: cmd/validatorcmd/getBalance.go [hunk 3]
```diff
@@ -58,68 +71,137 @@ func getBalance(_ *cobra.Command, _ []string) error {
...
+	case l1ListOption:
+		chainSpec := contract.ChainSpec{
+			BlockchainName: l1,
 		}
-		bootstrapValidatorsString := []string{}
-		bootstrapValidatorsToIndexMap := make(map[string]int)
-		for index, validator := range bootstrapValidators {
-			bootstrapValidatorsString = append(bootstrapValidatorsString, validator.NodeID)
-			bootstrapValidatorsToIndexMap[validator.NodeID] = index
+		chainSpec.SetEnabled(true, false, false, false, true)
```
Comment (arturrez) [inline cmd/validatorcmd/getBalance.go:130]: this is very cryptic line. not sure what it means `true, false, false, false, true`

Comment (arturrez) [reply cmd/validatorcmd/getBalance.go:130]: looking at 
```
func (cs *ChainSpec) SetEnabled(
	blockchainNameFlagEnabled bool,
	cChainFlagEnabled bool,
	pChainFlagEnabled bool,
	xChainFlagEnabled bool,
	blockchainIDFlagEnabled bool,
) {
	cs.blockchainNameFlagEnabled = blockchainNameFlagEnabled
	cs.cChainFlagEnabled = cChainFlagEnabled
	cs.pChainFlagEnabled = pChainFlagEnabled
	cs.xChainFlagEnabled = xChainFlagEnabled
	cs.blockchainIDFlagEnabled = blockchainIDFlagEnabled
}
```
seems to be not that useful and more confusing when used.

Comment (arturrez) [reply cmd/validatorcmd/getBalance.go:130]: pls feel free to ignore this

File: cmd/validatorcmd/getBalance.go [hunk 4]
```diff
@@ -58,68 +71,137 @@ func getBalance(_ *cobra.Command, _ []string) error {
...
+			true,
+			false,
+		)
 		if err != nil {
-			return err
+			return ids.Empty, false, err
 		}
-	} else {
-		validationID, err := app.Prompt.CaptureID("What is the validator's validationID?")
+		fmt.Println(rpcURL)
```
Comment (arturrez) [inline cmd/validatorcmd/getBalance.go:192]: debug?

File: cmd/validatorcmd/list.go [hunk 5]
```diff
@@ -50,6 +53,14 @@ func list(_ *cobra.Command, args []string) error {
 	}
 
 	blockchainName := args[0]
+	sc, err := app.LoadSidecar(blockchainName)
```
Comment (arturrez) [inline cmd/validatorcmd/list.go:56]: we should move this code up to the top of the func. including `blockchainName := args[0]`.
these is no point of doing networkoptions.GetNetworkFromCmdLineFlags if sidecar reports blockchain as not SoV

File: pkg/application/app.go [hunk 6]
```diff
@@ -620,7 +623,8 @@ func (app *Avalanche) GetBlockchainNamesOnNetwork(network models.Network) ([]str
 				}
 			}
 		}
-		if sc.Networks[networkName].BlockchainID != ids.Empty {
+		sovKindCriteria := !onlySOV || onlySOV && sc.Sovereign
```
Comment (arturrez) [inline pkg/application/app.go:626]: `!onlySOV || onlySOV` always returns `true`

Comment (arturrez) [reply pkg/application/app.go:626]: did you mean `!onlySOV || sc.Sovereign` ?

**Review by arturrez — APPROVED** [review]

File: cmd/validatorcmd/getBalance.go (hunk 1 above)
Comment (felipemadero) [reply cmd/validatorcmd/getBalance.go:82]: ignoring as in the other PR

File: cmd/validatorcmd/getBalance.go (hunk 3 above)
Comment (felipemadero) [reply cmd/validatorcmd/getBalance.go:130]: those 2 are not public

Comment (felipemadero) [reply cmd/validatorcmd/getBalance.go:130]: adding a comment to improve legibility

File: pkg/application/app.go (hunk 6 above)
Comment (felipemadero) [reply pkg/application/app.go:626]: && has precedence over ||, this is the same as "!onlySOV || (onlySOV && sc.Sovereign)"

**Pull request merged**